docs/build_conf.md

```diff
@@ -19,23 +19,80 @@ There following are some of the options to choose when building the device:
   TLS = openssl #underlying cryptography library to use. (`openssl` denotes the OpenSSL* toolkit.)
   MODULE = false #whether to use Secure Device Onboard (SDO) service-info functionality
 ```
-The default configuration can be overridden by using more options in `make`.<br>
+The default configuration can be overridden by using more options in `cmake`.<br>
 
 ## Custom build
-The default configuration can be overridden by using more options in `make`.<br>
+The default configuration can be overridden by using more options in `cmake`.<br>
 For example, to build the `STM32F429ZI` device:
 - BUILD: Debug mode
 - DA: ECDSA-256
 - AES_MODE: CBC
 - KEX: Diffie-Hellman
 - PK_ENC: rsa (Default)
 ```shell
-$ make TARGET_OS=mbedos BOARD=NUCLEO_F429ZI BUILD=debug AES_MODE=cbc KEX=dh DA=ecdsa256
+$ cmake -DTARGET_OS=mbedos -DBOARD=NUCLEO_F429ZI -DBUILD=debug -DAES_MODE=cbc -DKEX=dh -DDA=ecdsa256 .
+$ make -j4
 ```
 
 For available build options:
 ```shell
-make help
+List of build modes:
+BUILD=debug           # Debug mode (default)
+BUILD=release         # Release mode
+
+List of supported TARGET_OS:
+TARGET_OS=linux       # (Default)
+TARGET_OS=mbedos      # (Mbed OS v5.9.14)
+
+List of supported boards (valid only when TARGET_OS=mbedos):
+BOARD=NUCLEO_F767ZI   # (When building for STM32F767ZI MCU)
+BOARD=NUCLEO_F429ZI   # (When building for STM32F429ZI MCU)
+
+List of key exchange options:
+KEX=dh                # use Diffie-Hellman key exchange mechanism during TO2
+KEX=asym              # use Asymmetric key exchange mechanism during TO2
+KEX=ecdh              # use Elliptic-curve Diffie–Hellman key exchange mechanism during TO2 (default)
+KEX=ecdh384           # use Elliptic-curve Diffie–Hellman 384 bit key exchange mechanism during TO2
+
+List of AES encryption modes:
+AES_MODE=ctr          # use Counter mode encryption during TO2 (default)
+AES_MODE=cbc          # use Code-Block-Chaining mode encryption during TO2
+
+List of Device Attestation options:
+DA=ecdsa256           # Use ECDSA P256 based device attestation(default)
+DA=ecdsa384           # Use ECDSA-P384 based device attestation
+DA=tpm20_ecdsa256     # Use ECDSA-P256 based device attestation with TPM2.0 support
+MANUFACTURER_TOOLKIT=true # Use ECDSA with supply chain tool(default)
+DA_FILE=pem           # only Use if ECDSA private keys are PEM encoded
+
+List of Public Key encoding/owner-attestation options:
+PK_ENC=rsa            # Use RSAMODEXP-RSA2048RESTR public key encoding
+PK_ENC=ecdsa          # Use ECDSA-X.509 based public key encoding (default)
+
+Underlying crypto library to be used:
+TLS=openssl           # (Linux default, not supported for other TARGET_OS)
+TLS=mbedtls           # (Mbed OS default, not supported for other TARGET_OS)
+CRYPTO_HW=true        # Use Secure element for some of the crypto operations

```

Comment:
Should we mention that the default value is false?

Reply:
this is showing only the possible build configs. doesn't exactly represent the selected values. 
That will be shown at compile time.

Reply:
Instead  "Use Secure element for some of the crypto operations  ", write true/false allowed, default=false